ledger/src/ui/src/bolos_ux.c

```diff
@@ -483,77 +278,34 @@ void bolos_ux_main(void) {
             break;
 
         case BOLOS_UX_BOOT_ONBOARDING:
-            // re apply settings in the L4 (ble, brightness, etc) after exiting
-            // application in case of wipe
-            screen_settings_apply();
-
-            // request animation when dashboard has finished displaying all the
-            // elements (after onboarding OR the first time displayed)
-            G_bolos_ux_context.dashboard_redisplayed = 1;
-
-            // avoid reperso is already onboarded to avoid leaking data through
-            // parameters due to user land call
-            if (os_perso_isonboarded()) {
-                G_bolos_ux_context.exit_code = BOLOS_UX_OK;
-                break;
-            }
-
-            io_seproxyhal_init();
-            USB_power(1);
-            screen_settings_apply();
-            screen_not_personalized_init();
-            sample_main();
+            G_bolos_ux_context.exit_code = handle_bolos_ux_boot_onboarding();
             break;
-        case BOLOS_UX_DASHBOARD:
-            // apply settings when redisplaying dashboard
-            screen_settings_apply();
-
-            // when returning from application, the ticker could have been
-            // disabled
-            io_seproxyhal_setup_ticker(100);
-            // Run first application once
 
-            if (autoexec) {
-                autoexec = 0;
-                run_first_app();
-            }
-            screen_dashboard_init();
+        case BOLOS_UX_DASHBOARD:
+            handle_bolos_ux_boot_dashboard();
```

Comment:
My understanding is that all the calls that show a new screen, like this one, set the field `exit_code_after_elements_displayed` on `G_bolos_ux_context`. This field is used in other places in `bolos_ux.c` to set the actual value of `exit_code`.  With that in mind, it felt misleading/redundant to set this field here, since the actual value returned is handled by the screen logic. What do you think?

Reply:
Good point, I had no clue. Let's just leave it as is then.